### ISSUE TITLE
taking minimum of return value yields an error

### ISSUE THREAD

**Issue by campospinto** [pyccel/pyccel#835]
**Describe the bug**
When the minimum of a single return variable is taken, a semantic error is yielded

**To Reproduce**
Provide code to reproduce the behavior:
```python
def f() -> 'int':
    i = 1
    b = min(i,2)
    return i
```

Provide the generated code, or the error message:
```
ERROR at annotation (semantic) stage
pyccel:
 |error [semantic]: test_bug.py| Cannot determine dtype of min call with inhomogeneous arguments : <pyccel.ast.builtins.PythonTuple object at 0x122a96190>
Pyccel has encountered syntax that has not been implemented yet. Please create an issue at https://github.com/pyccel/pyccel/issues and provide a small example of your problem. Do not forget to specify your target language
```

**Expected behavior**
The code runs in python and should probably not yield an error

**Language**
Fortran and C

**Additional context**
The function `f` above is meaningless, but `g` below (which is more meaningful) yields the same error:
```python
def g() -> 'int':
    i = 1
    i = min(i,2)
    return i
```
Interestingly, the function `h` below does *not* produce an error, although it is very similar:
```python
def h() -> 'int':
    i = 1
    b = min(i,2)
    return b+i
```

**Comment by campospinto** [comment]
ps - as a quick fix in my code, I just write `return i+0`

**Comment by EmilyBourne** [comment]
Fixed by #912 but waiting for a cleaner fix with tests from #836
